apps/taskmasters/flowlord/handler.go

```diff
@@ -0,0 +1,313 @@
+				delete(wCache[j.Workflow], v)
+			}
+		}
+	}
+
+	// add files based tasks
+
+	for _, f := range tm.files {
+		wPath := f.workflowFile
+		w, found := sts.Workflow[wPath]
+		if !found {
+			w = make(map[string]cEntry)
+			sts.Workflow[wPath] = w
+		}
+		k := pName(f.Topic(), f.Job())
+		ent := cEntry{
+			Schedule: []string{f.SrcPattern},
+			Child:    tm.getAllChildren(f.Topic(), f.workflowFile, f.Job()),
+		}
+		w[k] = ent
+
+		// remove entries from wCache
+		delete(wCache[f.workflowFile], k)
+		for _, child := range ent.Child {
+			for _, v := range strings.Split(child, " ➞ ") {
+				delete(wCache[f.workflowFile], v)
+			}
+		}
+	}
+
+	// Add non cron based tasks
+	for f, w := range wCache {
+		for _, v := range w {
+			k := pName(v.Topic(), v.Job())
+			// check for parents
+			for v.DependsOn != "" {
+				if t, found := wCache[f][v.DependsOn]; found {
+					k = v.DependsOn
+					v = t
+				} else {
+					break
+				}
+
+			}
+
+			children := tm.getAllChildren(v.Topic(), f, v.Job())
+			// todo: remove children from Cache
+			if _, found := sts.Workflow[f]; !found {
+				sts.Workflow[f] = make(map[string]cEntry)
+			}
+			warning := validatePhase(v)
+			if v.DependsOn != "" {
+				warning += "parent task not found: " + v.DependsOn
+			}
+			sts.Workflow[f][k] = cEntry{
+				Schedule: make([]string, 0),
+				Warning:  warning,
+				Child:    children,
+			}
+		}
+	}
+
+	w.Header().Add("Content-Type", "application/json")
+	b, _ := json.MarshalIndent(sts, "", "  ")
+	w.Write(b)
+}
+
+func (tm *taskMaster) refreshHandler(w http.ResponseWriter, _ *http.Request) {
+	files, err := tm.refreshCache()
+	w.Header().Add("Content-Type", "application/json")
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	v := struct {
+		Files   []string `json:",omitempty"`
+		Updated time.Time
+	}{
+		Files:   files,
+		Updated: tm.lastUpdate.UTC(),
+	}
+	b, _ := json.MarshalIndent(v, "", "  ")
+
+	w.Write(b)
+}
+
+type request struct {
+	Task string
+	Job  string
+	From string // start
+	To   string // end
+	At   string // single time
+	By   string // month | day | hour // default by day,
+
+	Meta     map[string]string
+	Template string // should pull from workflow if possible
+	Execute  bool
+}
+
+func (tm *taskMaster) Backloader(w http.ResponseWriter, r *http.Request) {
+	req := request{}
+	b, _ := io.ReadAll(r.Body)
+	if err := json.Unmarshal(b, &req); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	resp := tm.backload(req)
+	if resp.code >= 400 {
+		http.Error(w, resp.Status, resp.code)
+		return
+	}
+
+	if req.Execute {
+		resp.Status = "Executed: " + resp.Status
+		errs := appenderr.New()
+		for _, t := range resp.Tasks {
+			errs.Add(tm.producer.Send(t.Type, t.JSONBytes()))
+		}
+		if errs.ErrOrNil() != nil {
+			http.Error(w, "issue writing to producer "+errs.Error(), http.StatusInternalServerError)
+		}
+	} else {
+		resp.Status = "DRY RUN ONLY: " + resp.Status
```

Comment:
on a dry run, could we add the info string to the response. It could be quite useful to see the info string when testing.

Reply:
maybe the meta string could be useful too.

Reply:
The response includes a list of every task created.